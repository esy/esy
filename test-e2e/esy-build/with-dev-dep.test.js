// @flow

const os = require('os');
const path = require('path');
const helpers = require('../test/helpers');
const {test, isWindows, isMacos, isLinux} = helpers;

helpers.skipSuiteOnWindows('Needs investigation');

function makePackage(
  p,
  {
    name,
    dependencies = {},
    devDependencies = {},
  }: {
    name: string,
    dependencies?: {[name: string]: string},
    devDependencies?: {[name: string]: string},
  },
  ...items
) {
  return helpers.dir(
    name,
    helpers.packageJson({
      name: name,
      version: '1.0.0',
      license: 'MIT',
      esy: {
        buildsInSource: true,
        build: [helpers.buildCommand(p, '#{self.root / self.name}.js')],
        install: [
          `cp #{self.root / self.name}.cmd #{self.bin / self.name}.cmd`,
          `cp #{self.root / self.name}.js #{self.bin / self.name}.js`,
        ],
      },
      dependencies,
      devDependencies,
    }),
    helpers.dummyExecutable(name),
    ...items,
  );
}

describe(`Project with "devDependencies"`, () => {
  async function createTestSandbox() {
    const p = await helpers.createTestSandbox();
    await p.fixture(
      helpers.packageJson({
        name: 'withDevDep',
        version: '1.0.0',
        esy: {
          build: 'true',
        },
        dependencies: {
          dep: 'path:./dep',
        },
        devDependencies: {
          devDep: 'path:./devDep',
        },
      }),
      makePackage(p, {
        name: 'dep',
        devDependencies: {devDepOfDep: '*'},
      }),
      makePackage(p, {
        name: 'devDep',
        dependencies: {
          depOfDevDep: 'path:../depOfDevDep',
        },
      }),
      makePackage(p, {
        name: 'depOfDevDep',
      }),
    );
    await p.esy('install');
    await p.esy('build');
    return p;
  }

  it('package "dep" should be visible in all envs', async () => {
    const p = await createTestSandbox();
    const expecting = expect.stringMatching('__dep__');

    {
      const {stdout} = await p.esy('dep.cmd');
      expect(stdout.trim()).toEqual(expecting);
    }

    {
      const {stdout} = await p.esy('b dep.cmd');
      expect(stdout.trim()).toEqual(expecting);
    }

    {
      const {stdout} = await p.esy('x dep.cmd');
      expect(stdout.trim()).toEqual(expecting);
    }
  });

  it(`package "dev-dep" is visible in command env / test env and via 'esy b CMD'`, async () => {
    const p = await createTestSandbox();

    {
      const {stdout} = await p.esy('devDep.cmd');
      expect(stdout.trim()).toEqual('__devDep__');
    }

    {
      const {stdout} = await p.esy('x devDep.cmd');
      expect(stdout.trim()).toEqual('__devDep__');
    }

    {
      const {stdout} = await p.esy('b devDep.cmd');
      expect(stdout.trim()).toEqual('__devDep__');
    }
  });

  test('build-env', async function() {
    const p = await createTestSandbox();
    const id = JSON.parse((await p.esy('build-plan')).stdout).id;
    const depId = JSON.parse((await p.esy('build-plan -p dep')).stdout).id;
    const devdepId = JSON.parse((await p.esy('build-plan -p devDep')).stdout).id;
    const depofdevdepId = JSON.parse((await p.esy('build-plan -p depOfDevDep')).stdout)
      .id;

    const {stdout} = await p.esy('build-env --json');
    const env = JSON.parse(stdout);
    expect(env).toMatchObject({
      cur__dev: 'true',
      cur__version: '1.0.0',
      cur__toplevel: `${p.projectPath}/_esy/default/store/i/${id}/toplevel`,
      cur__target_dir: `${p.projectPath}/_esy/default/store/b/${id}`,
      cur__stublibs: `${p.projectPath}/_esy/default/store/i/${id}/stublibs`,
      cur__share: `${p.projectPath}/_esy/default/store/i/${id}/share`,
      cur__sbin: `${p.projectPath}/_esy/default/store/i/${id}/sbin`,
      cur__root: `${p.projectPath}`,
      cur__original_root: `${p.projectPath}`,
      cur__name: `withDevDep`,
      cur__man: `${p.projectPath}/_esy/default/store/i/${id}/man`,
      cur__lib: `${p.projectPath}/_esy/default/store/i/${id}/lib`,
      cur__install: `${p.projectPath}/_esy/default/store/i/${id}`,
      cur__etc: `${p.projectPath}/_esy/default/store/i/${id}/etc`,
      cur__doc: `${p.projectPath}/_esy/default/store/i/${id}/doc`,
      cur__bin: `${p.projectPath}/_esy/default/store/i/${id}/bin`,
      PATH: [
        `${p.esyStorePath}/i/${depId}/bin`,
        `${p.esyStorePath}/i/${devdepId}/bin`,
        `${p.esyStorePath}/i/${depofdevdepId}/bin`,
        ``,
        `/usr/local/bin`,
        `/usr/bin`,
        `/bin`,
        `/usr/sbin`,
        `/sbin`,
      ].join(path.delimiter),
      OCAMLFIND_LDCONF: `ignore`,
      OCAMLFIND_DESTDIR: `${p.projectPath}/_esy/default/store/i/${id}/lib`,
      DUNE_BUILD_DIR: `${p.projectPath}/_esy/default/store/b/${id}`,
    });
  });

  test('build-env --release', async function() {
    const p = await createTestSandbox();
    const id = JSON.parse((await p.esy('build-plan --release')).stdout).id;
    const depId = JSON.parse((await p.esy('build-plan --release -p dep')).stdout).id;

    const {stdout} = await p.esy('build-env --json --release');
    const env = JSON.parse(stdout);
    expect(env).toMatchObject({
      cur__dev: 'false',
      cur__version: '1.0.0',
      cur__toplevel: `${p.projectPath}/_esy/default/store/i/${id}/toplevel`,
      cur__target_dir: `${p.projectPath}/_esy/default/store/b/${id}`,
      cur__stublibs: `${p.projectPath}/_esy/default/store/i/${id}/stublibs`,
      cur__share: `${p.projectPath}/_esy/default/store/i/${id}/share`,
      cur__sbin: `${p.projectPath}/_esy/default/store/i/${id}/sbin`,
      cur__root: `${p.projectPath}`,
      cur__original_root: `${p.projectPath}`,
      cur__name: `withDevDep`,
      cur__man: `${p.projectPath}/_esy/default/store/i/${id}/man`,
      cur__lib: `${p.projectPath}/_esy/default/store/i/${id}/lib`,
      cur__install: `${p.projectPath}/_esy/default/store/i/${id}`,
      cur__etc: `${p.projectPath}/_esy/default/store/i/${id}/etc`,
      cur__doc: `${p.projectPath}/_esy/default/store/i/${id}/doc`,
      cur__bin: `${p.projectPath}/_esy/default/store/i/${id}/bin`,
      PATH: [
        `${p.esyStorePath}/i/${depId}/bin`,
        ``,
        `/usr/local/bin`,
        `/usr/bin`,
        `/bin`,
        `/usr/sbin`,
        `/sbin`,
      ].join(path.delimiter),
      OCAMLFIND_LDCONF: `ignore`,
      OCAMLFIND_DESTDIR: `${p.projectPath}/_esy/default/store/i/${id}/lib`,
      DUNE_BUILD_DIR: `${p.projectPath}/_esy/default/store/b/${id}`,
    });
  });

  test('command-env', async function() {
    const p = await createTestSandbox();
    const id = JSON.parse((await p.esy('build-plan')).stdout).id;
    const depId = JSON.parse((await p.esy('build-plan -p dep')).stdout).id;
    const devDepId = JSON.parse((await p.esy('build-plan -p devDep')).stdout).id;
    const depOfDevDepId = JSON.parse((await p.esy('build-plan -p depOfDevDep')).stdout)
      .id;

    const {stdout} = await p.esy('command-env --json');
    const env = JSON.parse(stdout);

    const PATH = env.PATH.split(path.delimiter);

    expect(PATH).toContain(`${p.esyStorePath}/i/${depId}/bin`);
    expect(PATH).toContain(`${p.esyStorePath}/i/${devDepId}/bin`);
    expect(PATH).toContain(`${p.esyStorePath}/i/${depOfDevDepId}/bin`);

    expect(env).toMatchObject({
      cur__version: '1.0.0',
      cur__toplevel: `${p.projectPath}/_esy/default/store/i/${id}/toplevel`,
      cur__target_dir: `${p.projectPath}/_esy/default/store/b/${id}`,
      cur__stublibs: `${p.projectPath}/_esy/default/store/i/${id}/stublibs`,
      cur__share: `${p.projectPath}/_esy/default/store/i/${id}/share`,
      cur__sbin: `${p.projectPath}/_esy/default/store/i/${id}/sbin`,
      cur__root: `${p.projectPath}`,
      cur__original_root: `${p.projectPath}`,
      cur__name: `withDevDep`,
      cur__man: `${p.projectPath}/_esy/default/store/i/${id}/man`,
      cur__lib: `${p.projectPath}/_esy/default/store/i/${id}/lib`,
      cur__install: `${p.projectPath}/_esy/default/store/i/${id}`,
      cur__etc: `${p.projectPath}/_esy/default/store/i/${id}/etc`,
      cur__doc: `${p.projectPath}/_esy/default/store/i/${id}/doc`,
      cur__bin: `${p.projectPath}/_esy/default/store/i/${id}/bin`,
      OCAMLFIND_LDCONF: `ignore`,
      OCAMLFIND_DESTDIR: `${p.projectPath}/_esy/default/store/i/${id}/lib`,
      DUNE_BUILD_DIR: `${p.projectPath}/_esy/default/store/b/${id}`,
    });
  });

  test.enableIf(isMacos || isLinux)(
    'macos || linux: build-env snapshot',
    async function() {
      const p = await createTestSandbox();
      const id = JSON.parse((await p.esy('build-plan')).stdout).id;
      const depid = JSON.parse((await p.esy('build-plan -p dep')).stdout).id;
      const devdepid = JSON.parse((await p.esy('build-plan -p devDep')).stdout).id;
      const depofdevdepid = JSON.parse((await p.esy('build-plan -p depOfDevDep')).stdout)
        .id;
      const {stdout} = await p.esy('build-env --build-concurrency 16');
      expect(
        p.normalizePathsForSnapshot(stdout, {id, depid, devdepid, depofdevdepid}),
      ).toMatchSnapshot();
    },
  );

  test('build-env dep', async function() {
    const p = await createTestSandbox();
    const depId = JSON.parse((await p.esy('build-plan -p dep')).stdout).id;

    const {stdout} = await p.esy('build-env --json -p dep');
    const env = JSON.parse(stdout);
    expect(env).toMatchObject({
      cur__version: '1.0.0',
      cur__toplevel: `${p.esyStorePath}/s/${depId}/toplevel`,
      cur__target_dir: `${p.esyPrefixPath}/3/b/${depId}`,
      cur__stublibs: `${p.esyStorePath}/s/${depId}/stublibs`,
      cur__share: `${p.esyStorePath}/s/${depId}/share`,
      cur__sbin: `${p.esyStorePath}/s/${depId}/sbin`,
      cur__name: `dep`,
      cur__man: `${p.esyStorePath}/s/${depId}/man`,
      cur__lib: `${p.esyStorePath}/s/${depId}/lib`,
      cur__install: `${p.esyStorePath}/s/${depId}`,
      cur__etc: `${p.esyStorePath}/s/${depId}/etc`,
      cur__doc: `${p.esyStorePath}/s/${depId}/doc`,
      cur__bin: `${p.esyStorePath}/s/${depId}/bin`,
      PATH: [``, `/usr/local/bin`, `/usr/bin`, `/bin`, `/usr/sbin`, `/sbin`].join(
        path.delimiter,
      ),
      OCAMLFIND_LDCONF: `ignore`,
      OCAMLFIND_DESTDIR: `${p.esyStorePath}/s/${depId}/lib`,
    });
  });

  test.enableIf(isMacos || isLinux)(
    'macos || linux: build-env dep snapshot',
    async function() {
      const p = await createTestSandbox();
      const id = JSON.parse((await p.esy('build-plan -p dep')).stdout).id;
      const {stdout} = await p.esy('build-env -p dep --build-concurrency 16');
      expect(p.normalizePathsForSnapshot(stdout, {id})).toMatchSnapshot();
    },
  );

  test('build-env devDep', async function() {
    const p = await createTestSandbox();
    const devDepId = JSON.parse((await p.esy('build-plan -p devDep')).stdout).id;
    const depOfDevDepId = JSON.parse((await p.esy('build-plan -p depOfDevDep')).stdout)
      .id;

    const {stdout} = await p.esy('build-env --json -p devDep');
    const env = JSON.parse(stdout);
    expect(env).toMatchObject({
      cur__version: '1.0.0',
      cur__toplevel: `${p.esyStorePath}/s/${devDepId}/toplevel`,
      cur__target_dir: `${p.esyPrefixPath}/3/b/${devDepId}`,
      cur__stublibs: `${p.esyStorePath}/s/${devDepId}/stublibs`,
      cur__share: `${p.esyStorePath}/s/${devDepId}/share`,
      cur__sbin: `${p.esyStorePath}/s/${devDepId}/sbin`,
      cur__name: `devDep`,
      cur__man: `${p.esyStorePath}/s/${devDepId}/man`,
      cur__lib: `${p.esyStorePath}/s/${devDepId}/lib`,
      cur__install: `${p.esyStorePath}/s/${devDepId}`,
      cur__etc: `${p.esyStorePath}/s/${devDepId}/etc`,
      cur__doc: `${p.esyStorePath}/s/${devDepId}/doc`,
      cur__bin: `${p.esyStorePath}/s/${devDepId}/bin`,
      PATH: [
        `${p.esyStorePath}/i/${depOfDevDepId}/bin`,
        ``,
        `/usr/local/bin`,
        `/usr/bin`,
        `/bin`,
        `/usr/sbin`,
        `/sbin`,
      ].join(path.delimiter),
      OCAMLFIND_LDCONF: `ignore`,
      OCAMLFIND_DESTDIR: `${p.esyStorePath}/s/${devDepId}/lib`,
    });
  });

  test.enableIf(isMacos || isLinux)(
    'macos || linux: build-env devDep snapshot',
    async function() {
      const p = await createTestSandbox();
      const id = JSON.parse((await p.esy('build-plan -p devDep')).stdout).id;
      const depOfDevDepId = JSON.parse((await p.esy('build-plan -p depOfDevDep')).stdout)
        .id;
      const {stdout} = await p.esy('build-env -p devDep --build-concurrency 16');
      expect(p.normalizePathsForSnapshot(stdout, {id, depOfDevDepId})).toMatchSnapshot();
    },
  );

  test('exec-env', async function() {
    const p = await createTestSandbox();
    const id = JSON.parse((await p.esy('build-plan')).stdout).id;
    const depId = JSON.parse((await p.esy('build-plan -p dep')).stdout).id;
    const {stdout} = await p.esy('exec-env --json');
    const envpath = JSON.parse(stdout).PATH.split(path.delimiter);
    expect(
      envpath.includes(`${p.projectPath}/_esy/default/store/i/${id}/bin`),
    ).toBeTruthy();
    expect(envpath.includes(`${p.esyStorePath}/i/${depId}/bin`)).toBeTruthy();
  });

  test('command-env', async function() {
    const p = await createTestSandbox();
    const id = JSON.parse((await p.esy('build-plan')).stdout).id;
    const depId = JSON.parse((await p.esy('build-plan -p dep')).stdout).id;
    const devDepId = JSON.parse((await p.esy('build-plan -p devDep')).stdout).id;
    const {stdout} = await p.esy('command-env --json');
    const env = JSON.parse(stdout);
    expect(env).toMatchObject({
      cur__version: '1.0.0',
      cur__toplevel: `${p.projectPath}/_esy/default/store/i/${id}/toplevel`,
      cur__target_dir: `${p.projectPath}/_esy/default/store/b/${id}`,
      cur__stublibs: `${p.projectPath}/_esy/default/store/i/${id}/stublibs`,
      cur__share: `${p.projectPath}/_esy/default/store/i/${id}/share`,
      cur__sbin: `${p.projectPath}/_esy/default/store/i/${id}/sbin`,
      cur__root: `${p.projectPath}`,
      cur__original_root: `${p.projectPath}`,
      cur__name: `withDevDep`,
      cur__man: `${p.projectPath}/_esy/default/store/i/${id}/man`,
      cur__lib: `${p.projectPath}/_esy/default/store/i/${id}/lib`,
      cur__install: `${p.projectPath}/_esy/default/store/i/${id}`,
      cur__etc: `${p.projectPath}/_esy/default/store/i/${id}/etc`,
      cur__doc: `${p.projectPath}/_esy/default/store/i/${id}/doc`,
      cur__bin: `${p.projectPath}/_esy/default/store/i/${id}/bin`,
      OCAMLFIND_LDCONF: `ignore`,
      OCAMLFIND_DESTDIR: `${p.projectPath}/_esy/default/store/i/${id}/lib`,
      DUNE_BUILD_DIR: `${p.projectPath}/_esy/default/store/b/${id}`,
    });
    const envpath = env.PATH.split(path.delimiter);
    expect(envpath.includes(`${p.esyStorePath}/i/${depId}/bin`)).toBeTruthy();
    expect(envpath.includes(`${p.esyStorePath}/i/${devDepId}/bin`)).toBeTruthy();
  });
});

describe('Project with "devDependencies" (with "buildDev" config at the root)', () => {
  async function createTestSandbox() {
    const p = await helpers.createTestSandbox();
    const name = 'withDevDep';
    await p.fixture(
      helpers.packageJson({
        name: name,
        version: '1.0.0',
        esy: {
          build: [
            'cp #{self.name}.js #{self.target_dir / self.name}.js',
            helpers.buildCommand(p, '#{self.target_dir / self.name}.js'),
          ],
          // run commands from "devDependencies" here
          buildDev: [
            'devDep.cmd',
            'cp #{self.name}-dev.js #{self.target_dir / self.name}.js',
            helpers.buildCommand(p, '#{self.target_dir / self.name}.js'),
          ],
          install: [
            `cp #{self.target_dir / self.name}.cmd #{self.bin / self.name}.cmd`,
            `cp #{self.target_dir / self.name}.js #{self.bin / self.name}.js`,
          ],
        },
        dependencies: {
          dep: 'path:./dep',
        },
        devDependencies: {
          devDep: 'path:./devDep',
        },
      }),
      helpers.dummyExecutable(name),
      helpers.dummyExecutable(`${name}-dev`),
      makePackage(p, {
        name: 'dep',
        devDependencies: {devDepOfDep: '*'},
      }),
      makePackage(p, {
        name: 'devDep',
        dependencies: {
          depOfDevDep: 'path:../depOfDevDep',
        },
      }),
      makePackage(p, {
        name: 'depOfDevDep',
      }),
    );
    await p.esy('install');
    return p;
  }

  it(`can be built with 'esy build'`, async () => {
    const p = await createTestSandbox();

    {
      const {stdout} = await p.esy('build');
      expect(stdout).toBe(`__devDep__${os.EOL}`);
    }

    {
      const {stdout} = await p.esy('x withDevDep.cmd');
      expect(stdout).toBe(`__devDep__${os.EOL}__withDevDep-dev__${os.EOL}`);
    }
  });

  it(`can be built with 'esy build --release'`, async () => {
    const p = await createTestSandbox();

    {
      const {stdout} = await p.esy('build --release');
      expect(stdout).toBe(``);
    }

    {
      const {stdout} = await p.esy('x --release withDevDep.cmd');
      expect(stdout).toBe(`__withDevDep__${os.EOL}`);
    }
  });
});
