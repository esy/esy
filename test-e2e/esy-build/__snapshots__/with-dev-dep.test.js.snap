// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project with "devDependencies" macos || linux: build-env dep snapshot 1`] = `
"# Build environment
# package:            dep@path:dep
# depspec:            dependencies(self)
# mode:               build
# envspec:            
# buildIsInProgress:  true
# includeBuildEnv:    true
# includeCurrentEnv:  false
# includeNpmBin:      false


#
# Built-in
#
export cur__jobs="8"
export cur__etc="%esyStorePath%/s/%id%/etc"
export cur__share="%esyStorePath%/s/%id%/share"
export cur__toplevel="%esyStorePath%/s/%id%/toplevel"
export cur__stublibs="%esyStorePath%/s/%id%/stublibs"
export cur__doc="%esyStorePath%/s/%id%/doc"
export cur__man="%esyStorePath%/s/%id%/man"
export cur__lib="%esyStorePath%/s/%id%/lib"
export cur__sbin="%esyStorePath%/s/%id%/sbin"
export cur__bin="%esyStorePath%/s/%id%/bin"
export cur__install="%esyStorePath%/s/%id%"
export cur__target_dir="%esyPrefixPath%/3/b/%id%"
export cur__original_root="%esyPrefixPath%/source/i/dep__5d502355"
export cur__root="%esyPrefixPath%/3/b/%id%"
export cur__dev="false"
export cur__version="1.0.0"
export cur__name="dep"
export OCAMLFIND_LDCONF="ignore"
export OCAMLFIND_DESTDIR="%esyStorePath%/s/%id%/lib"
export SHELL="env -i /bin/bash --norc --noprofile"
export PATH="$PATH:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin""
`;

exports[`Project with "devDependencies" macos || linux: build-env devDep snapshot 1`] = `
"# Build environment
# package:            devDep@path:devDep
# depspec:            dependencies(self)
# mode:               build
# envspec:            
# buildIsInProgress:  true
# includeBuildEnv:    true
# includeCurrentEnv:  false
# includeNpmBin:      false


#
# Built-in
#
export cur__jobs="8"
export cur__etc="%esyStorePath%/s/%id%/etc"
export cur__share="%esyStorePath%/s/%id%/share"
export cur__toplevel="%esyStorePath%/s/%id%/toplevel"
export cur__stublibs="%esyStorePath%/s/%id%/stublibs"
export cur__doc="%esyStorePath%/s/%id%/doc"
export cur__man="%esyStorePath%/s/%id%/man"
export cur__lib="%esyStorePath%/s/%id%/lib"
export cur__sbin="%esyStorePath%/s/%id%/sbin"
export cur__bin="%esyStorePath%/s/%id%/bin"
export cur__install="%esyStorePath%/s/%id%"
export cur__target_dir="%esyPrefixPath%/3/b/%id%"
export cur__original_root="%esyPrefixPath%/source/i/devdep__3fd7f986"
export cur__root="%esyPrefixPath%/3/b/%id%"
export cur__dev="false"
export cur__version="1.0.0"
export cur__name="devDep"

#
# depOfDevDep@path:depOfDevDep@d41d8cd9
#
export CAML_LD_LIBRARY_PATH="%esyStorePath%/i/%depOfDevDepId%/stublibs:%esyStorePath%/i/%depOfDevDepId%/lib/stublibs:$CAML_LD_LIBRARY_PATH"
export OCAMLPATH="%esyStorePath%/i/%depOfDevDepId%/lib:$OCAMLPATH"
export MAN_PATH="%esyStorePath%/i/%depOfDevDepId%/man:$MAN_PATH"
export PATH="%esyStorePath%/i/%depOfDevDepId%/bin:$PATH"

#
# Built-in
#
export OCAMLFIND_LDCONF="ignore"
export OCAMLFIND_DESTDIR="%esyStorePath%/s/%id%/lib"
export SHELL="env -i /bin/bash --norc --noprofile"
export PATH="$PATH:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin""
`;

exports[`Project with "devDependencies" macos || linux: build-env snapshot 1`] = `
"# Build environment
# package:            withDevDep@link-dev:./package.json
# depspec:            dependencies(self)+devDependencies(self)
# mode:               buildDev
# envspec:            
# buildIsInProgress:  true
# includeBuildEnv:    true
# includeCurrentEnv:  false
# includeNpmBin:      false


#
# Built-in
#
export cur__jobs="8"
export cur__etc="%projectPath%/_esy/default/store/i/%id%/etc"
export cur__share="%projectPath%/_esy/default/store/i/%id%/share"
export cur__toplevel="%projectPath%/_esy/default/store/i/%id%/toplevel"
export cur__stublibs="%projectPath%/_esy/default/store/i/%id%/stublibs"
export cur__doc="%projectPath%/_esy/default/store/i/%id%/doc"
export cur__man="%projectPath%/_esy/default/store/i/%id%/man"
export cur__lib="%projectPath%/_esy/default/store/i/%id%/lib"
export cur__sbin="%projectPath%/_esy/default/store/i/%id%/sbin"
export cur__bin="%projectPath%/_esy/default/store/i/%id%/bin"
export cur__install="%projectPath%/_esy/default/store/i/%id%"
export cur__target_dir="%projectPath%/_esy/default/store/b/%id%"
export cur__original_root="%projectPath%"
export cur__root="%projectPath%"
export cur__dev="true"
export cur__version="1.0.0"
export cur__name="withDevDep"

#
# depOfDevDep@path:depOfDevDep@d41d8cd9
#
export CAML_LD_LIBRARY_PATH="%esyStorePath%/i/%depofdevdepid%/stublibs:%esyStorePath%/i/%depofdevdepid%/lib/stublibs:$CAML_LD_LIBRARY_PATH"
export OCAMLPATH="%esyStorePath%/i/%depofdevdepid%/lib:$OCAMLPATH"
export MAN_PATH="%esyStorePath%/i/%depofdevdepid%/man:$MAN_PATH"
export PATH="%esyStorePath%/i/%depofdevdepid%/bin:$PATH"

#
# devDep@path:devDep@d41d8cd9
#
export CAML_LD_LIBRARY_PATH="%esyStorePath%/i/%devdepid%/stublibs:%esyStorePath%/i/%devdepid%/lib/stublibs:$CAML_LD_LIBRARY_PATH"
export OCAMLPATH="%esyStorePath%/i/%devdepid%/lib:$OCAMLPATH"
export MAN_PATH="%esyStorePath%/i/%devdepid%/man:$MAN_PATH"
export PATH="%esyStorePath%/i/%devdepid%/bin:$PATH"

#
# dep@path:dep@d41d8cd9
#
export CAML_LD_LIBRARY_PATH="%esyStorePath%/i/%depid%/stublibs:%esyStorePath%/i/%depid%/lib/stublibs:$CAML_LD_LIBRARY_PATH"
export OCAMLPATH="%esyStorePath%/i/%depid%/lib:$OCAMLPATH"
export MAN_PATH="%esyStorePath%/i/%depid%/man:$MAN_PATH"
export PATH="%esyStorePath%/i/%depid%/bin:$PATH"

#
# Built-in
#
export OCAMLFIND_LDCONF="ignore"
export OCAMLFIND_DESTDIR="%projectPath%/_esy/default/store/i/%id%/lib"
export DUNE_BUILD_DIR="%projectPath%/_esy/default/store/b/%id%"
export DUNE_STORE_ORIG_SOURCE_DIR="true"
export SHELL="env -i /bin/bash --norc --noprofile"
export PATH="$PATH:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin""
`;
